plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -640,11 +640,11 @@ enterComputeCek = computeCek (toWordArray 0) where
     -- s ; ρ ▻ abs α L  ↦  s ◅ abs α (L , ρ)
     -- s ; ρ ▻ con c  ↦  s ◅ con c
     -- s ; ρ ▻ builtin bn  ↦  s ◅ builtin bn arity arity [] [] ρ
-    computeCek !unbudgetedSteps !ctx !_ term@(Builtin _ bn) = do
+    computeCek !unbudgetedSteps !ctx !_ (Builtin _ bn) = do
         !unbudgetedSteps' <- stepAndMaybeSpend BBuiltin unbudgetedSteps
         meaning <- lookupBuiltin bn ?cekRuntime
-        -- The @term@ is a 'Builtin', so it's fully discharged.
-        returnCek unbudgetedSteps' ctx (VBuiltin bn term meaning)
+        -- The term is a 'Builtin', so it's fully discharged.
+        returnCek unbudgetedSteps' ctx (VBuiltin bn (Builtin () bn) meaning)
```

Comment:
why not generalise ann here also?

Reply:
That thing goes out from the machine (during discharging), which is why it has to be `()` there, if we agree we don't need to return annotated terms, which would be problematic, because we'd need to annotate each result of a builtin application and it's not particularly clear how to synthesize an annotation in that case.

Reply:
We can put a `void` in before we throw it as we do in `discharge`, no? It's also potentially even performance relevant: this forces us to take apart and reconstruct a value instead of just passing through the program node.

Reply:
Hm, we could I guess, but we'd need to propagate annotations really far. So putting `ann` there would require us to also find an annotation in

```haskell
    applyEvaluate !unbudgetedSteps !ctx (VBuiltin fun term runtime) arg = do
        let argTerm = dischargeCekValue arg
            -- @term@ and @argTerm@ are fully discharged, and so @term'@ is, hence we can put it
            -- in a 'VBuiltin'.
            term' = Apply () term argTerm
```

which will require us to make `applyEvaluate` to take an annotation, which will require us to put an annotation into `FrameApplyFun`, which will require an annotation in `FrameApplyArg`.

All for removing them in the end anyway. So I'd rather not.

Reply:
... and the same for

```haskell
    forceEvaluate !unbudgetedSteps !ctx (VBuiltin fun term runtime) = do
        -- @term@ is fully discharged, and so @term'@ is, hence we can put it in a 'VBuiltin'.
        let term' = Force () term
```

Reply:
It's not on the very-hot evaluation path, because that field of `VBuiltin` is lazy. Which is why all that `Apply ()` and `Force ()` is fine (with the former being particularly expensive, since it discharges the argument, but we don't care since the field is lazy).

Plus, `void term` would reconstruct the term anyway, it's not that `void` magically changes the pointer to the annotation or something.

Reply:
Ah yes, good point about the laziness. I thought we were relying on laziness for the construction of the `Apply` stuff and so on, but yeah the `VBuiltin` field is also lazy. Unclear why it helps that it is, though, given that I'd expect the thing we put into it to be evaluated...

> Plus, void term would reconstruct the term anyway, it's not that void magically changes the pointer to the annotation or something.

I was unclear: I meant, leave the case of `computeCek` the same, and use `void` in `forceEvaluate`, because that code is only called on the error path.

Reply:
>  I thought we were relying on laziness for the construction of the `Apply` stuff and so on, but yeah the `VBuiltin` field is also lazy. 

It's the same field, `Apply ()` happens within this exact field of `VBuiltin`.

> I was unclear: I meant, leave the case of `computeCek` the same, and use `void` in `forceEvaluate`, because that code is only called on the error path.

Ah, I see.